android/app/src/main/java/app/organicmaps/util/Config.java

```diff
@@ -375,19 +376,37 @@ public static boolean isFirstLaunch(@NonNull Context context)
   public static void setFirstStartDialogSeen(@NonNull Context context)
   {
     MwmApplication.prefs(context)
-        .edit()
-        .putBoolean(KEY_MISC_FIRST_START_DIALOG_SEEN, true)
-        .apply();
+                  .edit()
+                  .putBoolean(KEY_MISC_FIRST_START_DIALOG_SEEN, true)
+                  .apply();
+  }
+
+  public static boolean isSearchHistoryEnabled()
```

Comment:
Let's make it more generic. There will also be the route history.
```suggestion
  public static boolean isHistoryEnabled()
```